testcases/kernel/syscalls/getrandom/getrandom02.c

```diff
@@ -0,0 +1,118 @@
+ * under the terms of version 2 of the GNU General Public License as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it would be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * Further, this software is distributed without any warranty that it is
+ * free of the rightful claim of any third person regarding infringement
+ * or the like.  Any license provided herein, whether implied or
+ * otherwise, applies only to this software file.  Patent licenses, if
+ * any, provided herein do not apply to combinations of this program with
+ * other software, or any other product whatsoever.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Contact information: Silicon Graphics, Inc., 1600 Amphitheatre Pkwy,
+ * Mountain View, CA  94043, or:
+ *
+ * http://www.sgi.com
+ *
+ * For further information regarding this notice, see:
+ *
+ * http://oss.sgi.com/projects/GenInfo/NoticeExplan/
+ */
+
+/*
+ * AUTHOR   : Cédric Hnyda
+ * DATE STARTED : 06/13/2015
+ *
+ *  Calls getrandom(2) with all the different possible flags, expects that 
+ *  the buffer is filled with random data and expects success.
+ */
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/fcntl.h>
+#include <errno.h>
+#include <string.h>
+#include <signal.h>
+#include <linux/random.h>
+#include <stdlib.h>
+#include <sys/syscall.h>
+#include <stdio.h>
+#include "test.h"
+#include "safe_macros.h"
+
+static int modes[] = {0, GRND_RANDOM, GRND_NONBLOCK,
+        GRND_RANDOM | GRND_NONBLOCK};
+
+char *TCID = "getrandom02";
+int TST_TOTAL = ARRAY_SIZE(modes);
+char *buf;
+size_t size = 256;
+
+
+void fill();
+int check_content(int nb);
+
+
+int main(int ac, char **av)
+{
+
+    int lc, i;
+    tst_parse_opts(ac, av, NULL, NULL);
+
+    for (lc = 0; TEST_LOOPING(lc); lc++) {
+        tst_count = 0;
+        for (i = 0; i < TST_TOTAL; i++) {
+            fill();
+            TEST(syscall(SYS_getrandom, buf, size, i));
+            if (!check_content(TEST_RETURN) && TEST_RETURN != -1)
+            {
+                tst_resm(TFAIL | errno, "getrandom failed");
+            }
+            else
+                tst_resm(TPASS, "getrandom returned %ld", TEST_RETURN);
+        }
+    }
+    if (buf)
+        free(buf);
+    tst_exit();
+}
+
+void fill()
+{
+    if (buf)
+        free(buf);
+    buf = malloc(sizeof (char) * size);
+    int i = 0;
+    for (i = 0; i < 100; i++)
+        buf[i] = '\0';
+}
+
+int check_content(int nb)
+{
+    int table[256];
+    int i, index, max;
+    for (i = 0; i < 256; ++i)
+        table[i] = 0;
+    max = nb * 0.10;
+    for (i = 0; i < nb; i++) {
+        index = (buf[i]) + 128;
```

Comment:
Hmm, we should define buf as unsigned char instead.
(If I remember right, C standard does not define if char is signed or not and leaves that to implementation so this expression is undefined anyway)